internal/handler/msg_saslcontinue.go

```diff
@@ -44,11 +44,18 @@ func (h *Handler) MsgSASLContinue(ctx context.Context, msg *wire.OpMsg) (*wire.O
 
 	conv := conninfo.Get(ctx).Conv()
 
+	if conv == nil {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrAuthenticationFailed,
+			"Authentication failed.",
+		)
```

Comment:
On CI, `test-integration-mongodb` fails differently, it returns `ProtocolError`.
```
  17.82s          	            	Diff:
  17.82s          	            	--- Expected
  17.82s          	            	+++ Actual
  17.82s          	            	@@ -1,6 +1,6 @@
  17.82s          	            	 (mongo.CommandError) {
  17.82s          	            	- Code: (int32) 18,
  17.82s          	            	- Message: (string) (len=22) "Authentication failed.",
  17.82s          	            	+ Code: (int32) 17,
  17.82s          	            	+ Message: (string) (len=27) "No SASL session state found",
  17.82s          	            	  Labels: ([]string) <nil>,
  17.82s          	            	- Name: (string) (len=20) "AuthenticationFailed",
  17.82s          	            	+ Name: (string) (len=13) "ProtocolError",
  17.82s          	            	  Wrapped: (error) <nil>,
  17.82s          	Test:       	TestAuthenticationSASLContinueFailure
```

Reply:
@chilagrow , yes. I'm going to take a look at why it is so, but this different failure is likely for a different ticket. I'm gonna check this once later today.

Reply:
Anything left, I will pick it up tomorrow :hugs:

Reply:
Chiming in here a bit because I see some possible things that can be addressed.

That protocol error is expected there, because in this [test](https://github.com/FerretDB/FerretDB/pull/4129/files/b2a023501384274ba3d010a37634a55d901227d9#diff-980d60c3913f30481316d120bf2a14061fb6349f514e7b99404513ca7df10a19R418) we have already established a [authenticated connection](https://github.com/FerretDB/FerretDB/pull/4129/files/b2a023501384274ba3d010a37634a55d901227d9#diff-980d60c3913f30481316d120bf2a14061fb6349f514e7b99404513ca7df10a19R443). We have completed a full SCRAM conversation in the driver. At step 7 it is improper and MongoDB will a return `ProtocolError` because no session state is ever created for this particular conversation.

Below are the sequence of operations from the test's output:

```
1. createUser OK
2. isMaster with helloOk OK so speculativeAuthenticate is done
3. saslStart
4. saslContinue payload Yz1iaXdz...
5. done true we are authenticated
6. ping OK
7. saslContinue payload WXoxaWFY... FAIL
```